doc/Learn/Tutorials/Authentication/HowTo-Cookies.md

```diff
@@ -3,11 +3,88 @@ uid: Learn.Tutorials.Authentication.HowToCookieAuthorization
+
+    ```csharp
+    protected override void OnLaunched(LaunchActivatedEventArgs args)
+    {
+        var builder = this.CreateBuilder(args)
+            .Configure(host => 
+            {
+                host.UseAuthentication(auth =>
+                    auth.AddCustom(custom =>
+                        custom.Login(
+                            async (sp, dispatcher, tokenCache, credentials, cancellationToken) =>
+                            {
+                                var isValid = credentials.TryGetValue("Username", out var username) && username == "Bob";
+                                return isValid ? 
+                                credentials : default;
+                            })
+                ));
+            });
+    ...
+    ```
+
+- Modify the app to add the `Cookies()` extension method. It configures the `IAuthenticationBuilder` to use cookies by registering a special HTTP request handler that will parse the response for tokens and store them in a cookie. It will apply them to future requests.
+
+    ```csharp
+    protected override void OnLaunched(LaunchActivatedEventArgs args)
+    {
+        var builder = this.CreateBuilder(args)
+            .Configure(host => 
+            {
+                host.UseAuthentication(auth =>
+                    auth.AddCustom(custom =>
+                        custom.Login(
+                            async (sp, dispatcher, tokenCache, credentials, cancellationToken) =>
+                            {
+                                var isValid = credentials.TryGetValue("Username", out var username) && username == "Bob";
+                                return isValid ? 
+                                credentials : default;
+                            })
+                ), 
+                configureAuthorization: builder =>
+                {
+                    builder
+                        .Cookies(/* options */);
+                });
+            });
+    ...
+    ```
+
+### 2. Configure cookie options
+
+- The `Cookies()` extension method takes two parameters; the first represents a name for the [access token](https://oauth.net/2/access-tokens/) cookie, and the second represents a name for the [refresh token](https://oauth.net/2/refresh-tokens/) cookie.
+
+    ```csharp
+    configureAuthorization: builder =>
+    {
+        builder
+            .Cookies("AccessToken", "RefreshToken");
+    ...
+    ```
+
+- Specifying a value for the latter is optional.
+
+### 3. Authorize with a token value from a cookie
+
+- Now, when you attempt to authenticate with a provider, the `Cookies()` extension method will attempt to authorize from a cookie by including any token information in the request. If the cookie is not found, it will attempt to authenticate with the provider as normal.
```

Comment:
I think we need more of an explanation for cookie handler in the context of auth:
If Cookie authorization is enabled, each service response will be used to extract access and refresh tokens. So, for example as part of the auth process, once user has logged in, there's typically a service call that retrieves access and refresh tokens - the response will be automatically parsed to extract these tokens from the cookies set on the response.
Subsequently, when making further service calls, the access and refresh tokens will be set as cookies on the request